File: typescript/solution.test.ts
import { fizzbuzz } from './solution';

describe('FizzBuzz', () => {
  test('returns array of 100 elements with correct values', () => {
    const result = fizzbuzz();
    
    expect(result.length).toBe(100);
    expect(result[0]).toBe("1");
    expect(result[2]).toBe("Fizz");
    expect(result[4]).toBe("Buzz");
    expect(result[14]).toBe("FizzBuzz");
  });
});

File: go/solution_test.go
package main

import (
	"reflect"
	"testing"
)

func TestFizzBuzz(t *testing.T) {
	result := FizzBuzz()

	expected := []string{
		"1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8", "Fizz", "Buzz",
		"11", "Fizz", "13", "14", "FizzBuzz",
	}

	if len(result) != 100 {
		t.Errorf("Expected 100 results, got %d", len(result))
	}

	if !reflect.DeepEqual(result[:15], expected) {
		t.Errorf("First 15 elements don't match. Got %v", result[:15])
	}
}
